scripts/build-run-test-dockerfile.sh

```diff
@@ -28,6 +28,50 @@ SERVICE_CONTROLLER_SOURCE_PATH=${SERVICE_CONTROLLER_SOURCE_PATH:-$DEFAULT_SERVIC
 SERVICE_CONTROLLER_E2E_TEST_PATH="${SERVICE_CONTROLLER_SOURCE_PATH}/test/e2e"
 
 KUBECONFIG_LOCATION="${KUBECONFIG:-"$HOME/.kube/config"}"
+AWS_CREDS_FILE_LOCATION="$HOME/.aws/credentials"
+
+# create an empty web-identity-token file inside Docker build context for
+# the test container.
+# If this script is running as part of prowjob then we will copy web-identity-token
+# in this file to be used inside test container
+pushd "$SCRIPTS_DIR" >/dev/null
+  touch web-identity-token
+popd >/dev/null
+
+# path of web-identity-token in test container
+TEST_CONTAINER_WEB_IDENTITY_TOKEN_FILE="/root/web-identity-token"
+# run python tests using new AWS profile to allow credential auto rotation
+ACK_TEST_AWS_PROFILE="ack-test"
+# If 'AWS_PROFILE' variable is set, use it as source profile for 'ack-test'
+# profile. Use 'default' as fallback
+ACK_TEST_SOURCE_AWS_PROFILE=${AWS_PROFILE:-"default"}
+# location of new credential file which will be copied in test container
+# Keep this file in same location as '~/.aws/credentials' file
+TEST_AWS_CREDS_FILE_LOCATION="$HOME/.aws/ack-test-credentials"
+
+if [[ -n $PROW_JOB_ID ]]; then
+  # If this is prowjob, create new aws credentials file and setup two profiles
+  # 1. 'prow-irsa' profile which gets aws credentials using web-identity-token
+  # 2. 'ack-test' profile which uses 'prow-irsa' as source profile and assumes
+  # ACK_ROLE_ARN for aws credentials
+  # NOTE: credentials in both these profiles rotate automatically
+
+  # copy web-identity-token file for use inside test container
+  cp "$AWS_WEB_IDENTITY_TOKEN_FILE" "$SCRIPTS_DIR"/web-identity-token >/dev/null
```

Comment:
I don't see where `$AWS_WEB_IDENTITY_TOKEN_FILE` is defined?

Reply:
AWS_WEB_IDENTITY_TOKEN_FILE variable is injected into pod using IRSA . I have added the comment explaining the same.